backend/danswer/connectors/document360/connector.py

```diff
@@ -0,0 +1,130 @@
+        self.api_token = credentials.get("document360_api_token")
+        self.portal_id = credentials.get("portal_id")
+        return None
+
+    def _make_request(self, endpoint: str, params: Optional[dict] = None) -> Any:
+        if not self.api_token:
+            raise ConnectorMissingCredentialError("Document360")
+
+        headers = {
+            'accept': "application/json",
+            'api_token': self.api_token
+        }
+
+        response = requests.get(f"{DOCUMENT360_API_BASE_URL}/{endpoint}", headers=headers, params=params)
+        response.raise_for_status()
+
+        return response.json()['data']
+
+    def _get_workspace_id_by_name(self):
+        projects = self._make_request("ProjectVersions")
+        return next((project['id'] for project in projects if project["version_code_name"] == self.workspace), None)
+    
+    def _get_articles_with_category(self, workspace_id):
+        all_categories = self._make_request(f"ProjectVersions/{workspace_id}/categories")
+        articles_with_category = []
+
+        for category in all_categories:
+            if category["name"] in self.categories or "ALL" in self.categories:
+                for article in category["articles"]:
+                    articles_with_category.append({
+                        "id": article["id"],
+                        "category_name": category["name"]
+                    })
+                for child_category in category["child_categories"]:
+                    for article in child_category["articles"]:
+                        articles_with_category.append({
+                            "id": article["id"],
+                            "category_name": child_category["name"]
+                        })
+        return articles_with_category
+
+    def _process_articles(
+        self, start: datetime | None = None, end: datetime | None = None
+    ) -> GenerateDocumentsOutput:
+        if self.api_token is None:
+            raise ConnectorMissingCredentialError("Document360")
+        
+        workspace_id = self._get_workspace_id_by_name()
+        articles = self._get_articles_with_category(workspace_id)
+
+        documents: List[Document] = []
+
+        for article in articles:
+            article_details = self._make_request(f"Articles/{article['id']}", {"langCode": "en"})
+            
+            updated_at = datetime.strptime(article_details["modified_at"], "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=None)
+            if start is not None and updated_at < start:
+                continue
+            if end is not None and updated_at > end:
+                continue
+            
+            doc_link = f"{DOCUMENT360_BASE_URL}/{self.portal_id}/document/v1/article/{article['id']}"
+            doc_text = f"workspace: {self.workspace}\n category: {article['category_name']}\n article: {article_details['title']} - {article_details.get('description', '')}"
```

Comment:
We definitely want to pull content and html_content and if either of those exist, parse those into the actual contents of the article